### ISSUE TITLE
Alternative to the animal names in alias_gensyms?

### ISSUE THREAD

**Issue by philtomson** [MikeInnes/MacroTools.jl#71]
I guess I'd like to have the option of having gensyms be converted to identifiers using some sort of deterministic naming scheme with minimal possibility for name collisions, like:

sym_id_1, sym_id_2, sym_id_3... etc. There are only so many animal names in animals.txt and I also worry that when the number of gensyms exceeds the number of animal names that this will lead to trouble.

**Comment by philtomson** [comment]
Ah, I see that there are 217 identifiers in animals.txt. If you run out of them then the pop! in the code below will fail:

```
function alias_gensyms(ex)
  left = copy(animals)
  syms = Dict{Symbol, Symbol}()
  prewalk(ex) do x
    isgensym(x) ? Base.@get!(syms, x, pop!(left)) : x
  end
end
```

**Comment by philtomson** [comment]
I've modified alias_gensyms to replace gensyms with unique identifiers. If you pass true as the 2nd parameter it uses the animal names:

```
"""
    alias_gensyms(expr)

Replaces gensyms with animal names if use_animals=true. 
Otherwise replaces gensyms with uniqe ids (deterministically)  
This makes gensym'd code far easier to follow.
"""
function alias_gensyms(ex, use_animals=false)
  counter = 0
  base_id = "sym_id_"
  left = copy(animals)
  syms = Dict{Symbol, Symbol}()
  prewalk(ex) do x
    if isgensym(x)
      repl_sym = Symbol("$base_id$counter")
      #tried the following, but it did not work:
      #repl_str = Symbol(replace(string(x), "#", ""))
      counter+=1
      if(use_animals)
        Base.@get!(syms, x, pop!(left))
      else
        Base.@get!(syms, x, repl_sym)
      end
    else
      x
    end
  end
end
```

For us this is helpful because we can compare output as we make changes to ONNX.jl.  Also, this would fix the problem where you have more than ~220 gensyms (currently that would lead to an error after you ran out of animals in animals.txt)

**Comment by MikeInnes** [comment]
Closed by #72
